### ISSUE TITLE
Conflict between ExplicitCollectionElementAccessMethod and UseOrEmpty when using SavedStateHandle in Android

### ISSUE THREAD

**Issue by mhernand40** [detekt/detekt#4791]
Please see the following code snippet:
```Kotlin
import androidx.lifecycle.SavedStateHandle

fun repro() {
  val savedStateHandle = SavedStateHandle()
  savedStateHandle["key"] = arrayListOf("a", "b", "c")
  savedStateHandle["key"] ?: emptyArray<String>()
}
```

## Expected Behavior
Detekt should not flag any violations for this because `savedStateHandle["key"]` returns `Any?`. The compiler cannot infer yet that it is an `ArrayList<String>`.

## Observed Behavior
Detekt complains with the following error:
```
This '?: emptyArray<String>()' can be replaced with 'orEmpty()' call [UseOrEmpty]
```
Then when attempting to apply the fix:
```Kotlin
import androidx.lifecycle.SavedStateHandle

fun repro() {
  val savedStateHandle = SavedStateHandle()
  savedStateHandle["key"] = arrayListOf("a", "b", "c")
  savedStateHandle.get<String>("key").orEmpty()
}
```
Detekt complains with the following error:
```
Prefer usage of the indexed access operator [] for map element access or insert methods. [ExplicitCollectionElementAccessMethod]
```

## Context
`SavedStateHandle` is a fairly common component in Android development. And it can store certain `Bundle` friendly types, arrays and array lists being one of them. However, when trying to extract the array or array list from the `Bundle`, these two Detekt rules conflict with one another. The only way I can see around this is to pick either approach from the snippets above and suppress the error.

## Your Environment

* Version of detekt used: 1.20.0
* Version of Gradle used (if applicable):
```
------------------------------------------------------------
Gradle 7.4.2
------------------------------------------------------------

Build time:   2022-03-31 15:25:29 UTC
Revision:     540473b8118064efcc264694cbcaa4b677f61041

Kotlin:       1.5.31
Groovy:       3.0.9
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          11.0.14.1 (Azul Systems, Inc. 11.0.14.1+1-LTS)
OS:           Mac OS X 12.3.1 aarch64
```
* Operating System and version: MacOS Monterey 12.3.1

**Comment by mhernand40** [comment]
I guess there is a workaround:
```Kotlin
import androidx.lifecycle.SavedStateHandle

fun repro() {
  val savedStateHandle = SavedStateHandle()
  savedStateHandle["key"] = arrayListOf("a", "b", "c")

  val valuesOrNull: ArrayList<String>? = savedStateHandle["key"]
  val values = valuesOrNull.orEmpty()
}
```
🤷‍♂️

**Comment by BraisGabin** [comment]
I think that you found 2 false positives at once.

- `bar.get<Foo>["hi"]` is not the same as `bar["hi"]`. So that's a false positive at `ExplicitCollectionElementAccessMethod`
- And you can't use `orEmpty` over classes that doesn't have that function. So `UseOrEmpty` should not complain about usages of `?: emptyArray<String>()` in those cases.

---

> I guess there is a workaround

For this kind of things I usually use `baseline` or `@Suppress`. Don't make your code uglier just because detekt tell you to do it.

**Comment by schalkms** [comment]
closed by [5f491f1](https://github.com/detekt/detekt/commit/5f491f1da607931f35e8d48cc6f91c7ebda63b6e)
